linker/linker_test.go

```diff
@@ -1638,6 +1676,32 @@ func TestLinkerValidation(t *testing.T) {
 					t.Errorf("expecting validation error %q; instead got: %q", tc.expectedErr, err)
 				}
 			}
+
+			// parse with protoc
+			var passProtoc bool
+			if tc.inputOrder != nil {
+				passProtoc, err = testByProtocWithFileOrder(t, tc.input, tc.inputOrder)
+			} else {
+				passProtoc, err = testByProtoc(t, tc.input)
```

Comment:
I think this condition check could be pushed down into `testByProtoc`. So just one function and always pass `tc.inputOrder` to it.